setup.py

```diff
@@ -36,6 +36,7 @@
     "fastjsonschema>=2.10",
     "python-constraint>=1.4",
     "python-dateutil>=2.8.0",
+    "scikit-learn>=0.20.0",
```

Comment:
I thought we were making scikit learn an optional dependency?

Reply:
Changed it to optional